### PR TITLE
feat(CO-46): GET /dungeon (dungeon instance)

### PR DESCRIPTION
**Pull request by rrtoth**

### CODE REVIEW

**Review by gbr88 — DISMISSED** [review]

File: src/services/dungeonService.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    this.equipmentService = equipmentService;
+  }
+
+  async newRandomDungeonInstance(heroId) {
+    const query = 'SELECT id FROM dungeons;';
+    return new Promise((resolve, reject) => {
+      this.conn.query(query, [], (err, row) => {
+        err ? reject(new Error(500)) : resolve(row);
+      });
+    }).then(async (res) => {
```
Comment (juhaszPeterDev) [inline src/services/dungeonService.js]: why do you need async in here?

Comment (rrtoth) [reply src/services/dungeonService.js]: accidentally left it there, now removed

**Review by gbr88 — DISMISSED** [review]

File: src/services/dungeonService.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      this.conn.query(query, [heroId], (err, res) => {
+        err ? reject(new Error(500)) : resolve(res[0]);
+      });
+    }).then((res) => {
+      if (res === undefined) {
+        return this.newRandomDungeonInstance(heroId);
+      }
+      return res;
+    }).then(async (res) => {
+      const dungeon = await this.getDungeonData(res.dungeonId); // 5
```
Comment (matyaspeterfi) [inline src/services/dungeonService.js]: left a comment :(

**Review by matyaspeterfi — CHANGES_REQUESTED** [review]

File: src/services/dungeonService.js (hunk 2 above)
Comment (rrtoth) [reply src/services/dungeonService.js]: deleted

**Review by matyaspeterfi — DISMISSED** [review]

**Review by juhaszPeterDev — APPROVED** [review]

**Pull request merged**